src/Cake.Core/ITaskTeardownContext.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+
+namespace Cake.Core
+{
+    public interface ITaskTeardownContext
+    {
+        ICakeTaskInfo Task { get; }
+        TimeSpan Duration { get; }
+        bool Skipped { get; }
+    }
+
+    public sealed class TaskTeardownContext : ITaskTeardownContext
+    {
+        public TaskTeardownContext(ICakeTaskInfo task, TimeSpan duration, bool skipped)
+        {
+            Task = task;
+            Duration = duration;
+            Skipped = skipped;
+        }
+
+        public ICakeTaskInfo Task { get; private set; }
```

Comment:
Use readonly backing fields that you assign in the constructor and provide only getters.